pkg/openstack/routes.go

```diff
@@ -180,22 +211,29 @@ func updateAllowedAddressPairs(network *gophercloud.ServiceClient, port *neutron
 
 // CreateRoute creates the described managed route
 func (r *Routes) CreateRoute(ctx context.Context, clusterName string, nameHint string, route *cloudprovider.Route) error {
-	klog.V(4).Infof("CreateRoute(%v, %v, %v)", clusterName, nameHint, route)
-
-	onFailure := newCaller()
+	r.Lock()
```

Comment:
do we have any criteria for this lock? 
e.g what the resource action should be lock/unlock? e.g all POST/UPDATE operation?

Reply:
@jichenjc gophercloud now supports atomic routes update. I submitted a new draft PR to use this feature: https://github.com/kubernetes/cloud-provider-openstack/pull/2134
It's based on this PR. Are there any updates on when this PR can be merged?